### ISSUE TITLE
Respect watch's WATCH_INTERVAL environment variable

### ISSUE THREAD

**Issue by benhoyt** [sachaos/viddy#45]
Nice tool! I usually set the `WATCH_INTERVAL` environment variable to 1 in my bashrc to give `watch` a default interval of 1 second. It would be nice if `viddy` respectd that variable too.

**Comment by coolabhays** [comment]
@benhoyt How would you like to make it work ? The flag for interval sets default interval to `2s` if you don't provide the interval flag.

Say, you set the `WATCH_INTERVAL` . There will be 2 things now:
1. Either the `-n` flag is provided. In this case, the value provided by flag will be used and not the value from env variable (doesn't matter if it is set or not)
2. You don't provide the flag. In that case, what you want to do ?

For the second case, a solution could be, check if env variable is set, if yes, then use the value from env variable. If no, then use the default value from program (which currently is 2s).

Is this the accepted behavior or do you have something else in mind ?

**Comment by benhoyt** [comment]
I think it should work the same as `watch` handles `WATCH_INTERVAL`: if `-n` is not provided, use `WATCH_INTERVAL`; if that's not provided, use the 2-second default. If `-n` is provided, it overrides `WATCH_INTERVAL`. I hacked together a version below that seems to work (though there may be a better way to automatically handle environment variables using Viper, I'm not sure):

```diff
$ git diff
diff --git a/config.go b/config.go
index 1f4f89a..f9799a7 100644
--- a/config.go
+++ b/config.go
@@ -3,6 +3,7 @@ package main
 import (
        "errors"
        "fmt"
+       "os"
        "strconv"
        "strings"
        "time"
@@ -72,7 +73,8 @@ func newConfig(v *viper.Viper, args []string) (*config, error) {
        flagSet := pflag.NewFlagSet("", pflag.ExitOnError)
 
        // runtimeConfig
-       flagSet.StringP("interval", "n", "2s", "seconds to wait between updates")
+       const defaultInterval = "2s"
+       flagSet.StringP("interval", "n", "", "seconds to wait between updates")
        flagSet.BoolP("precise", "p", false, "attempt run command in precise intervals")
        flagSet.BoolP("clockwork", "c", false, "run command in precise intervals forcibly")
        flagSet.BoolP("help", "h", false, "display this help and exit")
@@ -97,6 +99,12 @@ func newConfig(v *viper.Viper, args []string) (*config, error) {
        var conf config
 
        intervalStr, _ := flagSet.GetString("interval")
+       if intervalStr == "" {
+               intervalStr = os.Getenv("WATCH_INTERVAL")
+               if intervalStr == "" {
+                       intervalStr = defaultInterval
+               }
+       }
 
        interval, err := parseInterval(intervalStr)
        if err != nil {
diff --git a/main.go b/main.go
index 3f4eeba..ba32d56 100644
--- a/main.go
+++ b/main.go
@@ -74,7 +74,8 @@ Usage:
 Options:
   -b, --bell                 ring terminal bell changes between updates
   -d, --differences          highlight changes between updates
-  -n, --interval <interval>  seconds to wait between updates (default "2s")
+  -n, --interval <interval>  seconds to wait between updates (default
+                             WATCH_INTERVAL or "2s" if not set)
   -p, --precise              attempt run command in precise intervals
   -c, --clockwork            run command in precise intervals forcibly
   -t, --no-title             turn off header
```

**Comment by sachaos** [comment]
Thank to @coolabhays .
This feature is released in v0.3.5.
